### PR TITLE
[TEST] Server and Application classes

### PR DESCRIPTION
**Pull request by stevenzeiler**
Add high-level tests for the initial TLS server and the internal Express application.

[FOLD] Update language of Server test

[FOLD] Remove redundant initialization of supertest

### CODE REVIEW

File: test/application.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        done();
+      });
+  });
+
+  it.skip('should not upload an empty contract', function(done) {
+    http
+      .post('/contract')
+      .end(function(error, response) {
+        assert.strictEqual(response.body.success, false);
+        assert.strictEqual(response.body.error, 'no contract provided');
```
Comment (tdfischer) [inline test/application.js]: Not too sure if it makes sense to test what is meant to be a human-readable message.

Comment (stevenzeiler) [inline test/application.js]: Why not?

Comment (stevenzeiler) [inline test/application.js]: My goal is to at least be able to assert some sort of expected behavior. Right now the test will cause an exception.

Comment (stevenzeiler) [inline test/application.js]: I think we should have a spec that clearly identifies expected error cases so clients can handle them transparently.

Comment (tdfischer) [inline test/application.js]: End users of this should not be relying on human-friendly error messages. If we're testing this error message to confirm that the error conveys the fact that no contract was provided in the upload, there are other more machine friendly ways of doing it that are less prone to modification later on.

Not actually a problem, I'm just nitpicky.

**Comment by tdfischer** [comment]
No blockers, LGTM :+1:

File: test/application.js (hunk 1 above)
Comment (stevenzeiler) [inline test/application.js]: Alternatively we could design some error code map between integer or tokenized values and human-readable strings. I would say error messaging it is something to iterate on.

**Comment by emschwartz** [comment]
Merged.

**Comment by emschwartz** [comment]
LGTM too, except for the merge conflict

File: lib/application.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  app.set('compiler', engine.compiler);
+  app.set('fileManager', engine.fileManager);
+  app.set('engine', engine.engine);
+
+  app.get('/health', routeGetHealth);
+  app.post('/contract', routePostContract);
+  app.post('/token', routePostToken);
+  app.get('/token/:token', routeGetTokenMetadata.bind(null, Manager));
+
+  // TODO: take this out, obviously
```
Comment (stevenzeiler) [inline lib/application.js]: Let's remove this promptly before it makes it into `master`.

**Comment by codius-montecarlo** [comment]
The build is good and I'm seeing 2 +1s from the following folks:

* tdfischer* emschwartz

As such, I'll attempt to merge and close this pull request.

**Pull request merged**